lib/compress/zstd_compress.c

```diff
@@ -3280,22 +3333,27 @@ static size_t ZSTD_buildSeqStore(ZSTD_CCtx* zc, const void* src, size_t srcSize)
                 }
 
                 /* Fallback to software matchfinder */
-                {   ZSTD_blockCompressor const blockCompressor = ZSTD_selectBlockCompressor(zc->appliedParams.cParams.strategy,
-                                                                                            zc->appliedParams.useRowMatchFinder,
-                                                                                            dictMode);
+                {   ZSTD_blockCompressor const blockCompressor =
+                        ZSTD_selectBlockCompressor(
+                            zc->appliedParams.cParams.strategy,
+                            zc->appliedParams.useRowMatchFinder,
+                            dictMode);
                     ms->ldmSeqStore = NULL;
                     DEBUGLOG(
                         5,
                         "External sequence producer returned error code %lu. Falling back to internal parser.",
                         (unsigned long)nbExternalSeqs
                     );
+                    RETURN_ERROR_IF(blockCompressor == NULL, parameter_combination_unsupported, "Got NULL block compressor!");
```

Comment:
Question :
is this outcome `blockCompressor == NULL` possible,
meaning there is a combination of parameters that makes this situation reachable in production,
or is it essentially impossible, because it's part of the contract, and caller of `ZSTD_buildSeqStore()` are in charge of respecting it ?

Reply:
The first version of this functionality was designed around failing compressions that selected an unavailable compressor. I then changed the functionality to resolve unavailable selections down to an available implementation (in `adjustCParams()`). An `assert()` in `selectBlockCompressor()` remains active, validating that we never fail to resolve the cparams to an available implementation.

I though it might be harmless to leave these checks in here, but I guess it's not worth it. I'll go ahead and remove these.

Reply:
> I though it might be harmless to leave these checks in here, but I guess it's not worth it. I'll go ahead and remove these.

If by "harmless" you mean "no performance impact", then it's probably correct.

The comment was more targeted at understanding the code during a review: 
if there is an `if` branch, it's implied that both branches *could* be taken, given the proper circumstances.
Whereas, if it's an `assert()`, it means that this situation can never happen, and the code must ensure it.
This has consequences on the calling side (what invariants must be checked by the caller), on the contract documentation, and on the follow-up actions for the caller.

I don't remember the specifics, but I think I had in mind something along this line : " if `blockCompressor` can be `NULL`, then we return immediately here, but then, what are the consequences for the rest of the system ? Are they properly propagated ?".
Whereas, if it's not possible to happen, then there is no consequence to worry about.

Reply:
Indeed. For that reason, I've removed them.